packages/components/src/ProgressCircular/IndeterminateProgress.tsx

```diff
@@ -0,0 +1,214 @@
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import React, { FC } from 'react'
+import styled, { css, keyframes } from 'styled-components'
+import { progressCircularSVG, ProgressCircularSizes } from './size'
+import { progressCircularConstants } from './constants'
+import { CircleContainer } from './ProgressSvg'
+
+interface IndeterminateSpinnerProps {
+  size: ProgressCircularSizes
+}
+
+export const IndeterminateProgress: FC<IndeterminateSpinnerProps> = ({
+  size,
+}) => {
+  const { stroke, half, radius, dashArray, dashOffset } = progressCircularSVG({
+    size,
+  })
+
+  return (
+    <IndeterminateContainer>
+      <IndeterminateSpinner>
+        <CircleClipper>
+          <LeftCircleContainer
+            viewBox={`0 0 ${half * 2} ${half * 2}`}
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <circle
+              cx={half}
+              cy={half}
+              r={radius}
+              strokeDasharray={dashArray}
+              strokeDashoffset={dashOffset}
+              strokeWidth={stroke}
+            />
+          </LeftCircleContainer>
+        </CircleClipper>
+        <GapPatch>
+          <CircleContainer
+            viewBox={`0 0 ${half * 2} ${half * 2}`}
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <circle
+              cx={half}
+              cy={half}
+              r={radius}
+              strokeDasharray={dashArray}
+              strokeDashoffset={dashOffset}
+              strokeWidth={stroke - 0.6}
+            />
+          </CircleContainer>
+        </GapPatch>
+        <CircleClipper>
+          <RightCircleContainer
+            viewBox={`0 0 ${half * 2} ${half * 2}`}
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <circle
+              cx={half}
+              cy={half}
+              r={radius}
+              strokeDasharray={dashArray}
+              strokeDashoffset={dashOffset}
+              strokeWidth={stroke}
+            />
+          </RightCircleContainer>
+        </CircleClipper>
+      </IndeterminateSpinner>
+    </IndeterminateContainer>
+  )
+}
+
+const containerRotate = keyframes`
+  to {
+    transform: rotate(360deg);
+  }
+`
+
+const spinnerRotateAnimations = () => {
+  let animationString = ''
+  for (let i = 1; i < 9; i++) {
+    animationString += `${i * 12.5}% {
+      transform: rotate(${i * 0.5 * progressCircularConstants.arcSize}deg)
+    }`
+  }
+  return animationString
+}
+
+const spinnerKeyFrames = keyframes`
+  ${spinnerRotateAnimations()}
+`
```

Comment:
Just a small code style/best practices suggestion -- these styles could be rendered declaratively like so:

```suggestion
const spinnerRotateAnimations = () =>
  [...Array(9)].map(
    (_, i) =>
      `${i * 12.5}% {transform: rotate(${
        i * 0.5 * progressCircularConstants.arcSize
      }deg)}`
  )

const spinnerKeyFrames = keyframes`
  ${newSpinnerRotateAnimations().join('\n')}
`
```